andela/hc-aces-kla
# Ft telegram integrations 158174594

## Status
**READY**

## Migrations
YES

## Description
The system is able to send out alerts through telegram messages. Sets up health checks to be able to send out alerts concerning the cron jobs through telegram. Right now, Health checks only sends alerts through email, SMS, and voice.

## Related PRs
None

## Todos
- [x] Views
- [x] Checks UI


## Deploy Notes
On deployment run migrations ```./manage.py migrate```

Run background processes  ```./manage.py ensuretriggers``` and ```./manage.py sendalerts```

## Steps to Test or Reproduce

```sh
* After cloning the repo, run the application ```./ manage.py runserver```.
* Open the app in the browser, log in and add a check.
* Under the integrations tab, add the telegram integration. 
* Follow instructions and assign the check to it.
* Ping the check URL to change statuses and message will be sent
```

## Impacted Areas in Application
Created:
* add_telegram.html - Added UI for the page where the chat_id is added

Edited
* hc/api/transports.py - Added the telegram model
* hc/front/urls.py - Added the URL that redirects to the telegram pages
* hc/front/views.py - Added the function that loads the page where the chat_id is added
* hc/templates/front/channels.html - Added telegram to the list of configurable channel options
* hc/front/tests/test_add_channel.py - Added tests for telegram access
